lib/WeBWorK/lib/WebworkClient/classic_format.pl

```diff
@@ -4,33 +4,33 @@
 <head>
 <meta charset='utf-8'>
 <base href="$SITE_URL">
-<link rel="shortcut icon" href="/webwork2_files/images/favicon.ico"/>
+<link rel="shortcut icon" href="$SITE_HOST/webwork2_files/images/favicon.ico"/>
```

Comment:
Here, base href is `$SITE_URL` which is just an alias for `$ENV{baseURL}` which is used to construct `$webwork_htdocs_url` ...so this should be based off of `$webwork_htdocs_url` instead of `$SITE_HOST`

The base href doesn't work unless the renderedHTML is loaded into its own DOM document (own page or iframe), and generally, I think we should favor absolute URLs over relative ones... 

We have now: `baseURL` `SITE_HOST` and `formURL` -- they should be crafted in this way:
- `SITE_HOST` -- absolute URL, root endpoint of host (e.g. `https://myhost.com`)
- `baseURL` -- relative to `SITE_HOST`, extends to renderer root endpoint (e.g. `/webwork` as in your dev)
- `formURL` -- relative to `baseURL`, the renderer endpoint for problem submission (e.g. `/render-api`)

We also need to distinguish between these values as set in config, and their subsequent implementation as ENV variables. I think we construct `$ENV{baseURL}` as an absolute URL by concatenating with `SITE_HOST`. And then if any request overrides this, it is expected that the overriding value is an absolute URL. We should do the same for `$ENV{formURL}` -- which is where I can see more relevant use for overriding anyway.

Then, in these templates, use `$webwork_htdocs_url` instead of `$SITE_HOST/webwork2_files` -- that way the config file is the future entry point for making adjustments rather than having to modify each template.

Reply:
I believe this has now been accomplished for the renderer, though at a later time I need to add `$webwork_htdocs_url` to your Mojolicious application since it currently has hardcoded `/webwork2_files` in it.